packages/common/src/types/utility.ts

```diff
@@ -3,3 +3,5 @@ import { RecordWithId } from './index';
 export type ObjectWithStringKeys = Record<string, unknown>;
 
 export type RecordPatch<T extends RecordWithId> = Partial<T> & { id: string };
+
+export type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
```

Comment:
Better name? Making specific properties optional 🤔

Reply:
first thought was `PartialOn` which says the same thing. `PartialBy` is fine